pkg/cmd/experimental/syncgroups/config/types.go

```diff
@@ -0,0 +1,148 @@
+package config
+
+import (
+	ldaputil "github.com/openshift/origin/pkg/auth/authenticator/password/ldappassword/util"
+	"github.com/openshift/origin/pkg/client"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+// LDAPSyncCommonConfig contains all of the config that is common to all LDAP schemas
+type LDAPSyncCommonConfig struct {
+	// ServerConfig holds the configuration options for connecting with the LDAP server
+	ServerConfig ldaputil.LDAPServerConfig
+
+	// ProvisionGroups determines whether or not the GroupSyncer will create new OpenShift Groups if
+	// they do not yet exist or ignore those sync requests that do not map to existing OpenShift Groups
+	ProvisionGroups bool
+
+	// LDAPUserToIdentityMapping maps the attributes of a first-class LDAP user entry to an OpenShift
+	// Identity resource.
+	LDAPUserToIdentityMapping ldaputil.LDAPUserToIdentityMapping
+
+	// UserDefinedGroupNameMapping determines if the mapping of LDAP group UIDs to OpenShift Group
+	// names is defined by an input mapping or algorithmically
+	UserDefinedGroupNameMapping bool
+
+	// LDAPGroupUIDToOpenShiftNameMapping is an optional direct mapping of LDAP group UIDs to OpenShift
+	// Group names
+	LDAPGroupUIDToOpenShiftNameMapping map[string]string
+
+	// CLIOptions contains all of the options and references that are derived from the user input
+	CLIOptions LDAPSyncCommandLineOptions
+}
+
+// LDAPSyncCommandLineOptions contains all of the options and references that are derived from the user input
+type LDAPSyncCommandLineOptions struct {
```

Comment:
Why? I realize at this moment this is analogous to the `Options` struct in the cli file - one should remain, yes?